camunda-formio-tasklist-vue/src/services/SearchOptions.ts

```diff
@@ -144,54 +154,134 @@ export const FILTER_OPERATOR_TYPES = {
+    option[searchOption.key] = searchOption.value;
+  }
+  else if(searchOption.operator===FILTER_OPERATOR_TYPES.LIKE){
+    option[`${searchOption.key}Like`] = `%${searchOption.value}%`;
+  }else if(searchOption.operator===FILTER_OPERATOR_TYPES.BEFORE){
+    option[`${searchOption.key}Before`] = searchOption.value;
+  }else if(searchOption.operator===FILTER_OPERATOR_TYPES.AFTER){
+    option[`${searchOption.key}After`] = searchOption.value;
   }
+
+  return option;
+};
+
+export const isVariableTypeAvailable = (filterSelections: SearchOptionPayload[])=>{
+  return filterSelections.some(filter=>filter.type===FilterSearchTypes.VARIABLES);
 };
 
-export const getDeletedVariableIndex = (
-  deletequery: any,
-  selectetedSearchList: any,
-  key: string,
-  queryList: any
-) => {
-  for (let i = 0; i < queryList[key].length; i++) {
-    if (
-      queryList[key][i]["name"] === deletequery["name"]
-      && queryList[key]["value"] === deletequery["value"]
-    ) {
-      queryList[key].splice(i, 1);
+// export const searchValueObject = (searchValue: any, operator: any) => {
+//   if (operator === FILTER_OPERATOR_TYPES.EQUAL) {
+//     return searchValue;
+//   } else if (operator === FILTER_OPERATOR_TYPES.LIKE) {
+//     return `${searchValue}Like`;
+//   } else if (operator === FILTER_OPERATOR_TYPES.BEFORE) {
+//     return `${searchValue}Before`;
+//   } else if (operator === FILTER_OPERATOR_TYPES.AFTER) {
+//     return `${searchValue}After`;
+//   }
+// };
+
+// export const getDeletedVariableIndex = (
+//   deletequery: any,
+//   selectetedSearchList: any,
+//   key: string,
+//   queryList: any
+// ) => {
+//   for (let i = 0; i < queryList[key].length; i++) {
+//     if (
+//       queryList[key][i]["name"] === deletequery["name"]
+//       && queryList[key]["value"] === deletequery["value"]
+//     ) {
+//       queryList[key].splice(i, 1);
+//     }
+//   }
+//   return queryList;
+// };
+
+export const getFormattedQueryListParams = (searchOptionList: SearchOptionPayload[], searchQueryType: any, variableNameIgnoreCase: string, variableValueIgnoreCase: string)=>{
+  /*Function to transform the array of selected SearchQueries to a query object which can be used to search from TaskList*/
+  let resultList={
+  };
+  let paramList: any={
+  };
+  let isParamsHasValue=false;
+  if(searchOptionList.length===0){
+    return paramList;
+  }
+  paramList={
+    processVariables: [],taskVariables:[]
+  };
+
+  searchOptionList.forEach((searchOption)=>{
+    switch(searchOption.type){
+    case FilterSearchTypes.VARIABLES:
+      if(searchOption?.value && searchOption?.variable){
+        isParamsHasValue=true;
+        paramList[searchOption.key].push({
+          name: searchOption.variable, operator: getVariableOperator(searchOption.operator), value: searchOption.value
+        });
+      }
+      break;
+    case FilterSearchTypes.STRING:
+    case FilterSearchTypes.NORMAL:
+    case FilterSearchTypes.DATE:
```

Comment:
here i see, most of the cases are empty. then why switch case?.

Reply:
Used an if-else here